### PR TITLE
feat: split ruby hash and encryption rules

### PR DESCRIPTION
**Pull request by elsapet**
## Description

Splits Ruby weak encryption rule into several separate rules for each encryption / hashing library
Also makes use of @didroe 's suggestion here, to avoid repeating patterns simply for optional data type detection: https://github.com/Bearer/bearer/issues/870#issuecomment-1499115767

NOTE: with this change, the `ruby_lang_weak_encryption` rule is replaced by seven new rules: 

* `ruby_lang_weak_encryption_blowfish`
* `ruby_lang_weak_encryption_dsa`
* `ruby_lang_weak_encryption_rsa`
* `ruby_lang_weak_encryption_rc4`
* `ruby_lang_weak_hash_dss`
* `ruby_lang_weak_hash_md`
* `ruby_lang_weak_hash_sha`

Partially addresses #81

## Checklist

- [x] I've added a snapshot that shows my rule works as expected.
- [x] My rule has adequate metadata to explain its use.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: ruby/lang/weak_encryption_openssl_dsa.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    ```ruby
+    BCrypt::Password.create('iLOVEdogs123')
+    ```
+
+    ## Resources
+    - [BCrypt Explained](https://dev.to/sylviapap/bcrypt-explained-4k5c)
+  cwe_id:
+    - 331
+    - 326
+  id: ruby_lang_weak_encryption_openssl_dsa
```
Comment (didroe) [inline ruby/lang/weak_encryption_openssl_dsa.yml:55]: There's a bit of inconsistency with other rules, eg. `ruby_lang_weak_hash_md`. In the weak hash rule we have a mix of openssl and other patterns. Here we're naming it like we wouldn't add anything other than openssl.

I think we should either drop `openssl` from the name here, or split the other rule(s).

**Review by didroe — APPROVED** [review]

File: ruby/lang/weak_encryption_openssl_dsa.yml (hunk 1 above)
Comment (elsapet) [reply ruby/lang/weak_encryption_openssl_dsa.yml:55]: Good shout, thanks! Let's drop the `openssl` 👍

**Comment by cfabianski** [comment]
Do you want to rename `ruby_lang_weak_hash_md` to `ruby_lang_weak_hash_md5`

**Comment by elsapet** [comment]
@cfabianski the rule catches more than just MD5 cases: https://github.com/Bearer/bearer-rules/blob/main/ruby/lang/weak_hash_md.yml#L18 

For example,
```
OpenSSL::Digest::MD4.hexdigest("foo")
```

**Pull request merged**